README.md

```diff
@@ -608,6 +608,54 @@ buildable state (rather than be completely broken).
 
 ## Backend development
 
+### Code Style & Formatting
+
+.NET code style and formatting rules are imposed by [dotnet format](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-format) because it is IDE and platform agnostic, and natively included in the .NET SDK. Rule are dictated by an [.editorconfig file](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options).
```

Comment:
Can appreciate we've already thrown a bit of effort into `dotnet format`, but I've got to point out that the performance is pretty horrific.

Just operating on a single file takes at least 5+ seconds, and running across the entire repo takes at least 1m 30s on my machine. It's likely to be worse on lower spec machines.

For comparison, formatting with Prettier is basically instantaneous on individual files and can be ran across the project in around 15 seconds.

`dotnet format` could potentially be quite disruptive to general dev workflow, so we'll want to consider if this is within a tolerable threshold as a whole.

Would suggest that we potentially investigate alternatives like [CSharpier](https://csharpier.com/) as this is advertised to be fast. Just playing in the website's playground, the formatting on this is basically instantaneous.

It'd be good to compare the two!

Reply:
Yeah good shout - you're right that `dotnet format` is certainly slow! I haven't been worrying about it too much since I anticipate 99% of the formatting will be done by our IDEs as we go along, and it's only the PR pipelines that will pay the cost. I'll look at CSharpier and any other competitors I can find and do some comparison :)